packages/components/src/presentation-components/tree/DataProvider.ts

```diff
@@ -183,10 +183,12 @@ export class PresentationTreeDataProvider implements IPresentationTreeDataProvid
 
   /**
    * Returns a [NodeKey]($presentation-common) from given [TreeNodeItem]($components-react).
-   * **Warning:** the `node` must be created by this data provider.
+   *
+   * **Warning**: Returns invalid [NodeKey]($presentation-common) if `node` is not a [[PresentationTreeNodeItem]].
```

Comment:
I think we should also add the `@deprecated` tag here, in addition to adding it on the interface. At the moment it's not clear the method is deprecated when looking at this piece of code.